File: exercism/kotlin/easy/space-age/src/main/kotlin/SpaceAge.kt
class SpaceAge(private val seconds: Long) {

    fun onEarth() : Double = age(1.toDouble())
    fun onJupiter() : Double = age(11.862615)
    fun onMars() : Double = age(1.8808158)
    fun onMercury() : Double = age(0.2408467)
    fun onNeptune() : Double = age(164.79132)
    fun onSaturn() : Double = age(29.447498)
    fun onUranus() : Double = age(84.016846)
    fun onVenus() : Double = age(0.61519726)

    private fun age(orbital_period: Double) =
        "%.2f".format(seconds / orbital_period / 31557600).toDouble()        
}

File: exercism/csharp/easy/space-age/SpaceAge.cs
using System;

public class SpaceAge
{
    private long seconds;
    private double earthYearInSeconds = 31557600;
    public SpaceAge(long seconds)
    {
        this.seconds = seconds;
    }

    public double OnEarth() => seconds / earthYearInSeconds;

    public double OnMercury() => OnEarth() / 0.2408467;

    public double OnVenus() => OnEarth() / 0.61519726;

    public double OnMars() => OnEarth() / 1.8808158;

    public double OnJupiter() => OnEarth() / 11.862615;

    public double OnSaturn() => OnEarth() / 29.447498;

    public double OnUranus() => OnEarth() / 84.016846;

    public double OnNeptune() => OnEarth() / 164.79132;
}